lib/trie/database.go

```diff
@@ -185,22 +185,28 @@ func (t *Trie) loadNode(db Database, n *Node) error {
 	return nil
 }
 
-// PopulateMerkleValues writes the Merkle value of each children of the node given
-// as keys to the map merkleValues.
-func (t *Trie) PopulateMerkleValues(n *Node, merkleValues map[string]struct{}) {
-	if n.Kind() != node.Branch {
+// PopulateMerkleValues writes the Merkle values of the node given and of
+// all its descendant nodes as keys to the map merkleValues.
+// It is assumed the node and its descendant nodes have their Merkle value already
+// computed.
+func PopulateMerkleValues(n *Node, merkleValues map[string]struct{}) {
```

Comment:
would there be any value in storing the pointer to the `Node` in the map? `map[string]*Node`

Reply:
Nope :wink: 

This is (and will be) 100% only used for pruning which only cares about deleted and inserted trie node Merkle values (well, actually, I recently realized it's **trie node hashes**, since < 32B merkle values are inlined in parent node), which are the chaindb database keys (already now, and in the future for lazy loading).